causalpy/skl_meta_learners.py

```diff
@@ -0,0 +1,489 @@
+from copy import deepcopy
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from sklearn.linear_model import LogisticRegression
+from sklearn.utils import check_consistent_length
+from sklearn.model_selection import train_test_split
+
+from causalpy.utils import _is_variable_dummy_coded, _fit
+
+
+class MetaLearner:
+    "Base class for meta-learners."
+
+    def __init__(self, X: pd.DataFrame, y: pd.Series, treated: pd.Series) -> None:
+        # Check whether input is appropriate
+        check_consistent_length(X, y, treated)
+        if not _is_variable_dummy_coded(treated):
+            raise ValueError("Treatment variable is not dummy coded.")
+
+        self.cate = None
+        self.treated = treated
+        self.X = X
+        self.y = y
+        self.models = {}
+
+    def predict_cate(self, X: pd.DataFrame) -> np.array:
+        """
+        Predict out-of-sample conditional average treatment effect on given input X.
+        For in-sample treatement effect self.cate should be used.
+        """
+        raise NotImplementedError()
+
+    def predict_ate(self, X: pd.DataFrame) -> np.float64:
```

Comment:
What if we output the distribution instead of immediately returning the mean?

Reply:
Good idea, but maybe we should add a different method for that. I'd like it to be as similar to the sklearn counterpart as possible.